internal/monitors/conviva/metricconfig.go

```diff
@@ -0,0 +1,111 @@
+package conviva
+
+import (
+	"strings"
+)
+
+var metriclensMetrics = map[string][]string {
+	"quality_metriclens": {
+		"Total Attempts",
+		"Video Start Failures (%)",
+		"Exits Before Video Start (%)",
+		"Plays (%)",
+		"Video Startup Time (sec)",
+		"Rebuffering Ratio (%)",
+		"Average Bitrate (kbps)",
+		"Video Playback Failures (%)",
+		"Ended Plays",
+		"Connection Induced Rebuffering Ratio (%)",
+		"Video Restart Time",
+	},
+	"audience_metriclens": {
+		"Concurrent Plays",
+		"Plays",
+		"Ended Plays",
+	},
+}
+
+// MetricConfig for configuring individual metric
+type MetricConfig struct {
+	// Conviva customer account name. The default account is used if not specified.
+	Account                string   `yaml:"account"`
+	Metric                 string   `yaml:"metric" default:"quality_metriclens"`
+	// Filter names. The default is `All Traffic` filter
+	Filters                []string `yaml:"filters"`
+	// Metriclens dimension names.
+	MetriclensDimensions   []string `yaml:"metriclensDimensions"`
+	accountID              string
+	filterMap              map[string]string
+	metriclensDimensionMap map[string]float64
+	isInitialized          bool
+}
+
+func (m *MetricConfig) init(accountService *AccountService) {
+	if !m.isInitialized {
```

Comment:
Wrap this function in a mutex since it could possibly take longer than the metric interval to complete and should never be done twice concurrently.